lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java

```diff
@@ -0,0 +1,170 @@
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasket.getId())
+        .accessId(ACCESS_USER_1_1_ID)
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.READTASKS)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+  }
+
+  @WithAccessId(user = ACCESS_USER_1_1_ID)
+  @Test
+  void should_DeleteTaskComment_For_TaskCommentId() throws Exception {
+
+    taskService.deleteTaskComment(comment1.getId());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+  }
+
+  @WithAccessId(user = "user-1-2", groups = "user-1-1") // to read comments
+  @Test
+  void should_FailToDeleteTaskComment_When_UserHasNoAuthorization() throws Exception {
+    ThrowingCallable call = () -> taskService.deleteTaskComment(comment1.getId());
+    NotAuthorizedOnTaskCommentException e =
+        catchThrowableOfType(call, NotAuthorizedOnTaskCommentException.class);
+    assertThat(e.getTaskCommentId()).isEqualTo(comment1.getId());
+    assertThat(e.getCurrentUserId()).isEqualTo(taskanaEngine.getCurrentUserContext().getUserid());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(2);
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_DeleteTaskComment_When_UserIsInAdministrativeRole() throws Exception {
+    taskService.deleteTaskComment(comment1.getId());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_DeleteTaskComment_When_NoExplicitPermissionsButUserIsInAdministrativeRole()
+      throws Exception {
+
+    taskService.deleteTaskComment(comment1.getId());
+
+    // make sure the task comment was deleted
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+  }
+
+  @WithAccessId(user = ACCESS_USER_1_1_ID)
+  @Test
+  void should_FailToDeleteTaskComment_When_TaskCommentIdIsInvalid() throws Exception {
+    assertThatThrownBy(() -> taskService.deleteTaskComment(""))
+        .isInstanceOf(InvalidArgumentException.class);
+  }
+
+  @WithAccessId(user = ACCESS_USER_1_1_ID)
+  @Test
+  void should_FailToDeleteTaskComment_When_TaskCommentIdIsNull() throws Exception {
+    assertThatThrownBy(() -> taskService.deleteTaskComment(null))
+        .isInstanceOf(InvalidArgumentException.class);
+  }
+
+  @WithAccessId(user = ACCESS_USER_1_1_ID)
+  @Test
+  void should_FailToDeleteTaskComment_When_TaskCommentIsNotExisting() throws Exception {
```

Comment:
_When_CommentIdDoesntExist would be the correct test name

Reply:
Yes, I have replaced it